flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerRuntimeServices.java

```diff
@@ -83,7 +83,13 @@ private static SlotManager createSlotManager(
                     slotManagerConfiguration,
                     slotManagerMetricGroup,
                     new DefaultResourceTracker(),
-                    new FineGrainedTaskManagerTracker(),
+                    new FineGrainedTaskManagerTracker(
+                            slotManagerConfiguration.getMaxTotalCpu(),
+                            slotManagerConfiguration.getMaxTotalMem(),
+                            slotManagerConfiguration.isWaitResultConsumedBeforeRelease(),
+                            slotManagerConfiguration.getTaskManagerTimeout(),
+                            slotManagerConfiguration.getDeclareNeededResourceDelay(),
```

Comment:
How about introduce `TaskManagerTrackerConfiguration` and derive it from `SlotManagerConfiguration`.

Reply:
Good idea, but these configuration are used both in DeclarativeSlotManager and FineGrainedSlotManager. How about create a new issue to do this after we remove DeclarativeSlotManager?

Reply:
It makes sense to refactor this after `DeclarativeSlotManager` removed.